Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -421,5 +426,229 @@ lemma norm_le_interp_of_mem_verticalClosedStrip' (f : ℂ → E) {z : ℂ} {a b
+
+
+/--If z is on the closed strip `re ⁻¹' [l,u]`, then `(z-l)/(u-l)` is on the closed strip
+  `re ⁻¹' [0,1]`.-/
+lemma scale_mem_strip {z : ℂ} {l u : ℝ} (hul: l < u) (hz: z ∈ verticalClosedStrip l u) :
+    z/(u - l) - l/(u-l) ∈ verticalClosedStrip 0 1 := by
+  simp only [verticalClosedStrip, Complex.div_re, mem_preimage, sub_re, mem_Icc,
+    sub_nonneg, tsub_le_iff_right, ofReal_re, ofReal_im, sub_im, sub_self, mul_zero, zero_div,
+    add_zero]
+  simp only [verticalClosedStrip] at hz
+  norm_cast
+  simp_rw [Complex.normSq_ofReal, mul_div_assoc, div_mul_eq_div_div_swap,
+    div_self (by linarith : u - l ≠ 0), ← div_eq_mul_one_div]
+  constructor
+  · gcongr
+    · apply le_of_lt; simp [hul]
+    · exact hz.1
+  · rw [← sub_le_sub_iff_right (l / (u-l)), add_sub_assoc, sub_self, add_zero, div_sub_div_same,
+      div_le_one (by simp[hul]), sub_le_sub_iff_right l]
+    exact hz.2
+
+
+/-- The function `scale f l u` is `diffContOnCl`. -/
+lemma scale_diffContOnCl [NormedSpace ℂ E] {f: ℂ → E} {l u : ℝ} (hul: l < u)
+    (hd : DiffContOnCl ℂ f (verticalStrip l u)) : DiffContOnCl ℂ (scale f l u)
+    (verticalStrip 0 1) := by
+  unfold scale
+  apply DiffContOnCl.comp (s:= verticalStrip l u) hd
+  · apply DiffContOnCl.const_add
+    apply DiffContOnCl.smul_const
+    exact Differentiable.diffContOnCl differentiable_id'
+  · rw [MapsTo]
+    intro z hz
+    exact scale_mapsto_dom hul hz
+
+
+/-- The norm of the function `scale f l u` is bounded above on the closed strip `re⁻¹' [0, 1]`-/
+lemma scale_bddAbove {f: ℂ → E} {l u : ℝ} (hul: l< u)
+    (hB : BddAbove ((norm ∘ f) '' (verticalClosedStrip l u))) :
+    BddAbove ((norm ∘ (scale f l u)) '' (verticalClosedStrip 0 1)) := by
+  obtain ⟨R, hR⟩ := bddAbove_def.mp hB
+  rw [bddAbove_def]
+  use R
+  intro r hr
+  obtain ⟨w, hw₁, hw₂, _⟩ := hr
+  simp only [comp_apply, scale, smul_eq_mul]
+  have : ‖f (↑l + w * (↑u - ↑l))‖ ∈ norm ∘ f '' verticalClosedStrip l u := by
+    simp only [comp_apply, mem_image]
+    use ↑l + w * (↑u - ↑l)
+    simp only [and_true]
+    exact scale_mapsto_dom_cl hul hw₁
+  exact hR ‖f (↑l + w * (↑u - ↑l))‖ this
+
+
+/--A bound to the norm of `f` on the line `z.re=l` induces a bound to the norm of
+  `scale f l u z` on the line `z.re=0`. -/
+lemma scale_bound_left {f: ℂ → E} {l u a : ℝ} (ha : ∀ z ∈ re ⁻¹' {l}, ‖f z‖ ≤ a) :
+    ∀ z ∈ re ⁻¹' {0}, ‖scale f l u z‖ ≤ a := by
+  simp only [mem_preimage, mem_singleton_iff, scale, smul_eq_mul]
+  intro z hz
+  exact ha (↑l + z * (↑u - ↑l)) (by simp[hz])
+
+/--A bound to the norm of `f` on the line `z.re=u` induces a bound to the norm of `scale f l u z`
+  on the line `z.re=1`. -/
+lemma scale_bound_right {f: ℂ → E} {l u b : ℝ} (hb : ∀ z ∈ re ⁻¹' {u}, ‖f z‖ ≤ b) :
+    ∀ z ∈ re ⁻¹' {1}, ‖scale f l u z‖ ≤ b := by
+  simp only [scale, mem_preimage, mem_singleton_iff, smul_eq_mul]
+  intro z hz
+  exact hb (↑l + z * (↑u - ↑l)) (by simp[hz])
+
+/--A technical lemma needed in the proof-/
+lemma fun_arg_eq {l u: ℝ} (hul: l < u) (z: ℂ) :
```

Comment:
```suggestion
/--A technical lemma needed in the proof-/
private lemma fun_arg_eq {l u: ℝ} (hul: l < u) (z: ℂ) :
```
Let's make this a private lemma, something like this should definitely not appear in the docs.